tests/sniffer.py

```diff
@@ -7,36 +7,89 @@
+        outputfile = os.path.join(opts.tcpdump_dir, outputfile_basename) + ".pcap"
+        self.outputfiles[test_name] = outputfile
+        command = ["tcpdump", "-i", self.interface,'-U', '-w', outputfile]
+        self.tcpdump_proc = subprocess.Popen(command, stderr=subprocess.PIPE)
+        # tcpdump takes a while to start up. Wait for the appropriate output before continuing.
+        # poll() so we exit the loop if tcpdump terminates for any reason.
+        while not self.tcpdump_proc.poll():
+            line = self.tcpdump_proc.stderr.readline()
+            debug(line.decode()[:-1])  # strip off newline
+            if line.startswith(b"tcpdump: listening on " + self.interface.encode()):
+                # Make sure it doesn't block due to stderr buffering
+                self.tcpdump_proc.stderr.close()
+                break
+        else:
+            err("tcpdump terminated")
+            self.tcpdump_proc = None
+
+    def get_packet_capture(self, test_name):
+        if test_name in self.captures:
+            return self.captures[test_name]
+        tshark_command = ['tshark', '-r', self.outputfiles[test_name], '-T', 'json']
+        self.tshark_proc = subprocess.run(
+            tshark_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        try:
+            capture = json.loads(self.tshark_proc.stdout)
+            def ieee_filter(x): return 'ieee1905' in x['_source']['layers']
+            result = self._simplify_packets(filter(ieee_filter, capture))
+            if not result or len(result) == 0:
+                result = None
+            self.captures[test_name] = result
+            return self.captures[test_name]
+        except BaseException as ex:
+            return None
+
+    def _simplify_packets(self, packets):
+        def _simplify(x):
+            x['src_mac'] = x['_source']['layers']['eth']['eth.src']
+            x['dst_mac'] = x['_source']['layers']['eth']['eth.dst']
+            x['time_epoch'] = float(x['_source']['layers']['frame']['frame.time_epoch'])
+            x['time'] = x['_source']['layers']['frame']['frame.time']
+            x['ieee1905_layer'] = x['_source']['layers']['ieee1905']
+            x['msg_type'] = int(x['ieee1905_layer']['ieee1905.message_type'])
+            x['mid'] = int(x['ieee1905_layer']['ieee1905.message_id'], 16)
+            x['tlvs'] = {entry[0]: entry[1] for entry in filter(
+                lambda a: a[0].startswith('ieee1905.') == False, x['ieee1905_layer'].items())}
+            return x
+        return list(map(_simplify, packets))
+
+    def get_packets_for_cmdu_type(self, test_name, msg_type: int):
+        def filter_func(x): return x['msg_type'] == msg_type
+        try:
+            return list(filter(filter_func, self.get_packet_capture(test_name)))
+        except:
+            err("No packets captured for specified cmdu type: {}".format(msg_type))
+            return None
+        return list(filter(filter_func, self.get_packet_capture(test_name)))
```

Comment:
I think you must not be familiar with list comprehensions in python. They're really useful. 
```python
def get_packets_for_cmdu_type(self, test_name :str, msg_type: int):
    return [x for x in self.get_packet_capture(test_name) if x['msg_type'] == msg_type]
```